### PR TITLE
Use packaging.version in _compare_version

### LINKED ISSUE #10197: Use packaging.version for comparing versions
Because `distutils.Version` is deprecated, we have our own `_compare_version` function (https://github.com/mne-tools/mne-python/blob/main/mne/fixes.py#L32-L55). However, I was wondering (1) why this is in fixes and (2) why we do not use `packaging.version.Version` instead (which according to the `DeprecationWarning` is the recommended replacement). I'm happy to implement the changes (and probably move it back to `utils`?) if you think it makes sense. @hoechenberger WDYT?

Edit: Then we could also remove the [corresponding warning filter](https://github.com/mne-tools/mne-python/commit/523576f6bd805dfbf3e882933d779c5a2fd6d740#diff-d6b8de46e1e16287a7eed50a9cb2ed209e1d62ec5257722a2c0fce32dd2c726a).

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #10197. Do we want to move the function to a different module?

### CODE REVIEW

**Comment by cbrnr** [comment]
We have a circular import here, the problem is that in `check.py` we `from ..fixes import _median_complex`, and in `fixes.py` we `from .utils import _compare_version`.

We could nicely resolve this if we get rid of the `_median_complex` fix, which has already been fixed upstream I think (https://github.com/scipy/scipy/pull/12676/). However, I think this was released in SciPy 1.7, and we currently support >= 1.4.1. I think this fix has also been backported, but I'm not sure which versions received the fix. So should we bump the minimum SciPy version, and if yes, to which version? @larsoner

File: mne/utils/check.py [hunk 1]
```diff
@@ -17,10 +17,32 @@
...
+    version_b : str
+        Second version string.
+
+    Returns
+    -------
+    bool
+        The result of the version comparison.
+    """
+    from packaging.version import parse  # noqa:F401
+    return eval(f'parse("{version_a}") {operator} parse("{version_b}")')
```
Comment (larsoner) [inline mne/utils/check.py:43]: Please restore the `catch_warnings` wrapper I added in https://github.com/mne-tools/mne-python/pull/10199#discussion_r786996171 that you can also see in the [diff above](https://github.com/mne-tools/mne-python/pull/10223/files#diff-50629b846cfe5156a9681aa5a131f1511f177e420981d19db50b2e6c67816e3eL55-L56). We (and users) don't need to hear a warning if package versions are not PEP440 compatible when using this function.

**Comment by larsoner** [comment]
Let's move `_compare_version` back to `fixes.py` -- we don't want `fixes.py` to import from any other `mne` namespace for exactly this reason. It's nice to know everything in there is always safe to use with no risk of circular import

File: mne/utils/check.py (hunk 1 above)
Comment (cbrnr) [reply mne/utils/check.py:43]: I don't think `packaging.version.parse` raises this warning. Can you give an example of a problematic version and/or comparison?

Comment (larsoner) [reply mne/utils/check.py:43]:
```
>>> import warnings
>>> warnings.simplefilter('always', DeprecationWarning)
>>> import packaging.version
>>> packaging.version.parse('0.1.develop')
/home/larsoner/.local/lib/python3.10/site-packages/packaging/version.py:111: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
  warnings.warn(
<LegacyVersion('0.1.develop')>
```

Comment (cbrnr) [reply mne/utils/check.py:43]: Are `DeprecationWarnings` not shown by default (that's why I didn't see it)? Thanks for the explanation, I'll revert the change.

Comment (cbrnr) [reply mne/utils/check.py:43]: Apparently yes: https://docs.python.org/3/library/warnings.html#default-warning-filter

**Comment by cbrnr** [comment]
All green!

**Comment by hoechenberger** [comment]
Perfect, thanks, @cbrnr!

**Pull request merged**